docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+
+Given below is an example usage scenario of the goal feature.
+
+Step 1. The user launches the application for the first time. `Ui#printGoal` is called, 
+a line informing them they have no goal is printed.
+
+Step 2. The user executes `goal save money` command to set the goal as "save money".
+The `goal` command is passed through a parser to return the GoalCommand with arguments initialised.
+
+Step 3. `GoalCommand#execute()` is called, setting the goal in `UserData` to "save money". The change is printed
+with `Ui#printChangeGoal()`.
+
+Step 4. The user executes `goal` command to print the current goal.
+The `goal` command is passed through a parser to return the GoalCommand with blank arguments.
+
+Step 5.  `GoalCommand#execute()` is called, to retrieve the current goal in `UserData` and print it
+with `Ui#printGoal()`.
+
+The following sequence diagram shows how the `goal save money` command is parsed:
+
+![Sequence diagram for parsing goal command](./diagrams/GoalParseSequenceDiagram.jpg)
+
+The following sequence diagram shows how `GoalCommand#execute()` works:
+
+![Sequence diagram for goal command execute](./diagrams/GoalExecuteSequenceDiagram.jpg)
+
+#### Done feature
+(WIP)
+
+#### Undone feature
+(WIP)
+
+#### Delete feature
+(WIP)
+
+#### Note feature
+(WIP)
+
+#### Reminder feature
+(WIP)
+
+#### Extract feature
+(WIP)
+
+
+ 
+## Documentation, logging, testing, configuration, dev-ops
+
+## Appendix: Requirements
+
+### Product scope
+
+##### Target user profile
+
+<ul>
+    <li>has a need to organise events using scheduler</li>
+    <li>prefer desktop application</li>
+    <li>can type fast</li>
+    <li>prefer typing to mouse interactions</li>
+    <li>is reasonably comfortable using CLI apps</li>
+</ul>
+
+##### Value proposition
+
+Manage one's events quickly compared to GUI scheduling applications. 
+Users can check when they are free using a simple command 
+and also extract deadlines from any body of text.
+
+### User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|office worker (working from home)|organise large amount of online Zoom meetings to manage|keep track of his schedule for the day|
+|v1.0|university student|consolidate my schedule for both my online zoom lectures and my ordinary timetable|better manage my time|
+|v1.0|university student having projects|be informed of the goals of my project|keep track of the project details|
+|v1.0|university student|customise repeated task|set events like tutorials to weekly|
+|v1.0|teacher|organise all my zoom links for my classes|it can be easily accessible.|
+|v1.0|user|see the events that I have created|refer to them when I forget|
```

Comment:
I think it would be best to specify user of your app as user is a really broad term and can refer to a user of anything